lib/auth/auth_with_roles.go

```diff
@@ -4505,12 +4519,35 @@ func (a *ServerWithRoles) ReplaceRemoteLocks(ctx context.Context, clusterName st
 // channel if one is encountered. Otherwise the event channel is closed when the stream ends.
 // The event channel is not closed on error to prevent race conditions in downstream select statements.
 func (a *ServerWithRoles) StreamSessionEvents(ctx context.Context, sessionID session.ID, startIndex int64) (chan apievents.AuditEvent, chan error) {
-	if err := a.actionForKindSession(apidefaults.Namespace, types.VerbList, sessionID); err != nil {
+	createErrorChannel := func(err error) (chan apievents.AuditEvent, chan error) {
 		c, e := make(chan apievents.AuditEvent), make(chan error, 1)
 		e <- trace.Wrap(err)
 		return c, e
```

Comment:
Wouldn't this work? Do we need the left channel to be non-nil?
```suggestion
		e := make(chan error, 1)
		e <- trace.Wrap(err)
		return nil, e
```

Reply:
Yeah it needs to be non-nil, due to the way it's consumed - https://github.com/gravitational/teleport/blob/master/lib/auth/grpcserver.go#L3232-L3252

Reply:
A `nil` channel there works exactly as we expect, blocking and letting the branch on the error channel take the error and exit.